tests/appsec/test_automated_login_events.py

```diff
@@ -1218,3 +1218,55 @@ def assert_priority(span, meta):
         assert (
             meta["manual.keep"] == "true"
         ), 'meta.manual.keep should be "true" when _sampling_priority_v1 is not MANUAL_KEEP'
+
+
+@rfc("https://docs.google.com/document/d/19VHLdJLVFwRb_JrE87fmlIM5CL5LdOBv4AmLxgdo9qI/edit")
+@features.user_monitoring
+@scenarios.appsec_auto_events_rc
+class Test_V2_Login_Events_RC:
+    """
+    Sequence of events: (default) -> disabled
+    TODO add more uses cases when we have better semantics to control RC in the libs
+    """
+
+    USER = "test"
+    PASSWORD = "1234"
+
+    @property
+    def username_key(self):
+        """ In Rails the parametesr are group by scope. In the case of the test the scope is user. The syntax to group parameters in a POST request is scope[parameter] """
+        return "user[username]" if "rails" in context.weblog_variant else "username"
+
+    @property
+    def password_key(self):
+        """ In Rails the parametesr are group by scope. In the case of the test the scope is user. The syntax to group parameters in a POST request is scope[parameter] """
+        return "user[password]" if "rails" in context.weblog_variant else "password"
+
+    def _remote_config_is_applied(self, state_matcher):
+        def _wait_for_config(data):
+            if data["path"] == "/v0.7/config":
+                if "config_states" in data.get("request", {}).get("content", {}).get("client", {}).get("state", {}):
+                    config_states = data["request"]["content"]["client"]["state"]["config_states"]
+                    if state_matcher(config_states):
+                        return True
+            return False
+
+        return _wait_for_config
+
+    def setup_disabled(self):
+        def _is_disabled_state(config_states):
+            for s in config_states:
+                if s["id"] == "auto-user-instrum-disabled":
+                    return True
+            return False
+
+        interfaces.library.wait_for(self._remote_config_is_applied(_is_disabled_state), timeout=30)
+        self.request_disabled = weblog.post(
+            "/login?auth=local", data={self.username_key: self.USER, self.password_key: self.PASSWORD}
+        )
+
+    def test_disabled(self):
+        assert self.request_disabled.status_code == 200
+        for _, _, span in interfaces.library.get_spans(request=self.request_disabled):
```

Comment:
```suggestion
        spans = [s for _, _, s in interfaces.library.get_spans(request=self.request_disabled)]
        assert spans, "No spans to validate"
        for span in spans:
```

This will ensure that the test does not pass if something is terribly broken and there are no spans at all.

Reply:
Done, thanks for the suggestion :smile: